atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/keyvalue/api/cache/SnapshotStoreImplTest.java

```diff
@@ -114,22 +119,73 @@ public void removeTimestampOnlyRetentionsDownToMinimumSize() {
         assertSnapshotsEqualForTimestamp(SNAPSHOT_3, TIMESTAMP_3);
         assertSnapshotsEqualForTimestamp(SNAPSHOT_4, TIMESTAMP_4);
 
-        removeSnapshotAndAssert(TIMESTAMP_1, SEQUENCE_1).isEmpty();
-        removeSnapshotAndAssert(TIMESTAMP_2, SEQUENCE_2).isEmpty();
-        removeSnapshotAndAssert(TIMESTAMP_3, SEQUENCE_3).hasValue(SNAPSHOT_3);
-        removeSnapshotAndAssert(TIMESTAMP_4, SEQUENCE_4).hasValue(SNAPSHOT_4);
+        removeSnapshotAndAssert(TIMESTAMP_2, SEQUENCE_2)
+                .as("snapshot is the latest non-essential and thus kept")
+                .hasValue(SNAPSHOT_2);
+
+        removeSnapshotAndAssert(TIMESTAMP_1, SEQUENCE_1)
+                .as("snapshot is not the latest and thus removed")
+                .isEmpty();
+
+        removeSnapshotAndAssert(TIMESTAMP_4, SEQUENCE_4)
+                .as("snapshot is the latest non-essential and thus kept")
+                .hasValue(SNAPSHOT_4);
+
+        removeSnapshotAndAssert(TIMESTAMP_3, SEQUENCE_3)
+                .as("snapshot is not the latest and thus removed")
+                .isEmpty();
 
         snapshotStore.storeSnapshot(SEQUENCE_5, ImmutableSet.of(TIMESTAMP_5), SNAPSHOT_5);
-        removeSnapshotAndAssert(TIMESTAMP_5, SEQUENCE_5).hasValue(SNAPSHOT_5);
-        assertThat(snapshotStore.getSnapshotForSequence(SEQUENCE_3)).isEmpty();
+        removeSnapshotAndAssert(TIMESTAMP_5, SEQUENCE_5)
+                .as("snapshot is the latest non-essential and thus kept")
+                .hasValue(SNAPSHOT_5);
+
+        assertThat(snapshotStore.getSnapshotForSequence(SEQUENCE_4))
+                .as("snapshot is no longer latest and thus removed")
+                .isEmpty();
```

Comment:
I think we want a test here around things being removed when a long-standing transaction is still present. In other words, we have say transactions 1, 2, 3, 4 with snapshots A, B, C, D. I think we want remove 4 -> kept, remove 2 -> 2 gone, 4 still kept.